data/scenarios/Challenges/Ranching/gated-paddock/meandering-sheep.sw

```diff
@@ -0,0 +1,36 @@
+// A "sheep" that wanders around randomly.
+
+let forever : cmd unit -> cmd unit = \c. c ; forever c in
+let repeat : int -> cmd unit -> cmd unit =
+  \n. \c. if (n == 0) {} {c ; repeat (n-1) c} in
+let randdir : cmd dir =
+  d <- random 4;
+  return (
+    if (d == 0) {north}
+    {if (d == 1) {east}
+    {if (d == 2) {south} {west}}})
+  in
+
+forever (
+  n <- random 30;
+  wait (30 + n);
+  d <- randdir;
+  turn d;
+  dist <- random 3;
+  repeat dist (
+    b <- blocked;
+    if b {} {
+      move;
+
+      // Sheep can drown.
+      hasWater <- ishere "water";
+      if hasWater {
+        say "whoops!";
+        selfdestruct;
+      } {};
+
+    }
+  );
+  r <- random 5;
+  if (r == 0) { say "baaa" } {}
```

Comment:
I don't think there is any way to hear the baas.  Maybe install a `hearing aid` on the base robot?

Reply:
I think @kostmo fixed that. 🙂

Reply:
@xsebek now it seems like when sheep say "baaa", it flashes instantly in the Messages dialog, like for one tick, and then disappears.  I am not sure exactly how the Messages dialog is supposed to work re: `say` but having things disappear after one tick does not seem very useful.

Reply:
@byorgey you should still see it if you have a hearing device. The idea is that you should not hear things from before you were able to hear anything at all. So newly heard messages go to the logger from which they are shown in the Messages modal.

But I see it does not work now. Either @kostmo accidentally broke it, or I made a mistake in the `listen`/`say` implementation. 😄

Reply:
It's a bug. 🐛 Fix incoming.

Reply:
Fixed in #842. ✅